### ISSUE TITLE
[padding-line-between-tags] Support singleline/multiline options

### ISSUE THREAD

**Issue by Ky6uk** [vuejs/eslint-plugin-vue#1974]
**What rule do you want to change?**

`vue/padding-line-between-tags`

**Does this change cause the rule to produce more or fewer warnings?**

More warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

It could be an additional option like singleline/multiline.

```json
{ "blankLine": "always", "prev": "div-singleline", "next": "span-multiline" }
```

But it might be a bad example as `<div-singleline />` might be the correct tag/component. So it could be also:

```json
{ "blankLine": "always", "prevMulti": "div", "nextSingle": "span" }
```

**Please provide some example code that this change will affect:**

```html
<div>
  This is a multiline tag
<div>
<span>This is a singleline tag</span>
```

Will trigger an error with configuration like that:

```json
{ "blankLine": "always", "prevMulti": "div", "nextSingle": "span" }
```

Moreover, it will be possible to force the required padding line between multiline and singleline tags in general.

For the ruleset like this:

```json
[
  { "blankLine": "always", "prevMulti":  "*", "nextSingle": "*" },
  { "blankLine": "always", "prevSingle": "*", "nextMulti":  "*" },
  { "blankLine": "always", "prevMulti":  "*", "nextMulti":  "*" },
  { "blankLine": "never",  "prevSIngle": "*", "nextSingle": "*" }
]
```

It will be possible to have code like this:

```html
<div>
  This is a multiline tag
<div>

<div>
  This is a multiline tag
<div>

<div>
  This is a multiline tag
<div>

<span>This is a singleline tag</span>
<span>This is a singleline tag</span>

<div>
  This is a multiline tag
<div>

<span>This is a singleline tag</span>

<div>
  This is a multiline tag
<div>

<span>This is a singleline tag</span>
<span>This is a singleline tag</span>
<span>This is a singleline tag</span>
```

**Comment by FloEdelmann** [comment]
CC @dev1437 and @amiranagram, follow-up to #1832 and #1966.

**Comment by dev1437** [comment]
Should I just allow users to define their own regex? That would solve this problem and quite possibly future problems where people want more control. Thoughts? @FloEdelmann

**Comment by FloEdelmann** [comment]
I don't think a regex would be very readable for this use case, and also: [Don't parse HTML with regex](https://stackoverflow.com/a/1732454) :wink: 

I like the `prevMulti` / `prevSingle` / `nextMulti` / `nextSingle` options better. But I don't know if this use case is common enough to consider implementing it.

**Comment by wJoenn** [comment]
Hello there.
I'm resurrecting this thread a but just to say I've been wanting this option for years and never realised there was an opened issue to implement this.

Are you still willing to implement this option ? Are you open for a PR to be opened at some point if you haven't beforehand ?

**Comment by Ky6uk** [comment]
@wJoenn no. Feel free to implement this. I guess it's still a nice-to-have feature even nowadays.
